packages/opentelemetry-sdk-trace-base/test/common/export/BatchSpanProcessorBase.test.ts

```diff
@@ -175,43 +175,35 @@ describe('BatchSpanProcessorBase', () => {
       assert.strictEqual(spy.args.length, 0);
     });
 
-    it('should export the sampled spans with buffer size reached', done => {
-      const clock = sinon.useFakeTimers();
+    it('should export the sampled spans with buffer size reached', async () => {
       const processor = new BatchSpanProcessor(exporter, defaultBufferConfig);
-      for (let i = 0; i < defaultBufferConfig.maxExportBatchSize; i++) {
-        const span = createSampledSpan(`${name}_${i}`);
+      const span = createSampledSpan(name);
+      for (let i = 1; i < defaultBufferConfig.maxExportBatchSize; i++) {
```

Comment:
On a few tests you've changed these loops to create the span outside the loop and count up from 1 instead of 0. Is this just to optimize performance/allocations or is there some necessary purpose to that?

Reply:
These tests were all set up to expect the export to happen lazily, I changed to start from `1` so that these tests still get the single export they expect rather than two